packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -1592,6 +1653,16 @@ export namespace TreeWidget {
                         });
                     }
                 }}
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                onScroll={(e: any) => {
+                    const scrollTop = e.target.scrollTop;
+                    const scrollHeight = e.target.scrollHeight;
+                    const clientHeight = e.target.clientHeight;
+                    const isAtBottom = scrollHeight - scrollTop - clientHeight <= 30; // 30px threshold
```

Comment:
nit: I think it'd be more clear if we'd assign the 30 to a variable.
```suggestion
                    const thresholdInPx = 30;
                    const isAtBottom = scrollHeight - scrollTop - clientHeight <= thresholdInPx;
```